DependencyInjectionContainer/DIContainer.cs

```diff
@@ -38,9 +38,27 @@ public void Register(object implementation, ServiceLifetime lifetime)
             services.Add(new Service(implementation, lifetime));
         }
 
+        public TResolveType Resolve<TResolveType>()
+        {
+            return (TResolveType)Resolve(typeof(TResolveType));
+        }
 
+        public IEnumerable<TResolveType> ResolveMany<TResolveType>()
```

Comment:
In lines where I can't comment violated rule https://github.com/DotNetAnalyzers/StyleCopAnalyzers/blob/master/documentation/SA1516.md